public/js/home.js

```diff
@@ -1181,6 +1192,220 @@ function closeSideNav() {
+     * text is the main requirement, and it's just text
+     * color: String - string reference to a color or a variable, sets the background color
+     * textColor: String - string reference to a color or a variable, sets the text color
+     * buttonText: String - Sets the button text, both it and buttonClick have to be defined for the button to show
+     * buttonClick: Function - Sets the button's onclick logic, both it and buttonText have to be defined for the button to show
+     * destroyWhenButtonClicked : Boolean - Whether or not it should destroy itself when the button is clicked, defaults to true
+     * bodyClick: Function - Sets the body's onclick logic
+     * destroyWhenBodyClicked : Boolean - Whether or not it should destroy itself when the body is clicked, defaults to true
+     * timeout: Int - Time in ms  
+     * timeoutFunction: Function - What to run on timeout
+     */
+    constructor(text, options) {
+        this.text = text;
+        this.color = options["color"];
+        this.textColor = options["textColor"];
+        this.buttonText = options["buttonText"];
+        this.buttonClick = options["buttonClick"];
+        this.destroyWhenButtonClicked = options["destroyWhenButtonClicked"] || true;
+        this.bodyClick = options["bodyClick"];
+        this.destroyWhenBodyClicked = options["destroyWhenBodyClicked"] || true;
+
+        //timeout logic
+        const timeoutFunction = options["timeoutFunction"] === undefined ? options["timeoutFunction"] : () => {};
+        if (options["timeout"] !== undefined) {
+            setTimeout(() => {
+                timeoutFunction();
+                this.destroy();
+            }, options["timeout"]);
+        }
+
+        this.id;
+    }
+
+    /**
+     * creates the snackbar in the HTML
+     * if you want to show it as soon as you make it, use show without calling make instead
+     * returns the snackbar object
+     */
+    make() {
+        //stops if the element already exists
+        if (typeof document.getElementById(`sidenav-${this.id}`) === undefined) {
+            return;
+        }
+
+        //gives it an ID if it doesn't already have one
+        if (this.id === undefined) {
+            this.createID();
+        }
+
+        //creates the snackbar and gives it classes
+        const snackbarNode = document.createElement("DIV");
+        snackbarNode.classList.add("snackbar");
+        snackbarNode.classList.add("hidden");
+
+        //assigns it id based off of it's actual id
+        snackbarNode.id = `snackbar-${this.id}`;
+
+        //adds color if given
+        if (this.color !== undefined) {
+            snackbarNode.style.backgroundColor = `${this.color}`;
+        }
+
+        //sets the body onclick listener which just destroys it by default
+        const bodyOnClickFunction = this.bodyClick !== undefined ? () => {this.bodyClick();} : () => {};
+        const destroyFromBody = this.destroyWhenBodyClicked ? () => {this.destroy();} : () => {};
```

Comment:
Simplify arrow functions
```suggestion
        const bodyOnClickFunction = this.bodyClick !== undefined ? () => this.bodyClick() : () => {};
        const destroyFromBody = this.destroyWhenBodyClicked ? () => this.destroy() : () => {};
```